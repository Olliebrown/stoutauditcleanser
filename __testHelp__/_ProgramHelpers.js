import fs from 'fs'
import path from 'path'

// Project specific imports
import { QUERIES } from '../src/domTraversal/queriesAndRegex.js'
import Program from '../src/Objects/Program.js'

export function InitProgramTest (htmlFileName) {
  // Read in the HTML page
  const baseHTML = fs.readFileSync(
    path.join('__tests__', 'data', 'ProgramTables', htmlFileName),
    'utf8'
  )

  // Load page from imported HTML
  document.documentElement.innerHTML = baseHTML.toString()

  // Extract program headers and make sure there's at least one
  const headerNodes = document.querySelectorAll(QUERIES.programHeader)
  if (headerNodes?.length < 1) {
    throw new Error('Unable to find at least one program')
  }

  // Create program object
  return new Program(headerNodes[0])
}
