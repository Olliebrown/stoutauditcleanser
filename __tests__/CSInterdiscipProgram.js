// Necessary when using ECMA modules
import { jest } from '@jest/globals'

// Help load the program HTML page
import { InitProgramTest } from '../__testHelp__/_ProgramHelpers.js'

// Disable jest mocking for fs module
jest.dontMock('fs')

describe('BS in CS: Interdisciplinary Program Node', () => {
  // Inject clean version of HTML page and rebuild program before tests
  let program = null
  beforeAll(() => {
    // Load page from imported HTML
    program = InitProgramTest('CSInterdiscipProgram.html')
  })

  // Clear out the HTML and program
  afterAll(() => {
    jest.resetModules()
    program = null
  })

  // Basic page structure tests
  test('should have a specific page title and specific program prefix', () => {
    // Check title
    const title = document.querySelector('title')
    expect(title).toBeTruthy()
    expect(title.textContent).toBe('BS in Computer Science Interdisciplinary')

    // Check program name
    expect(program).toBeTruthy()
    expect(program.getName()).toMatch(/^BS In CS - Interdisciplinary\s+\d\d\d\d$/)
  })

  test('should have expected snapshot', () => {
    expect(program?.toJSON()).toMatchSnapshot()
  })
})
