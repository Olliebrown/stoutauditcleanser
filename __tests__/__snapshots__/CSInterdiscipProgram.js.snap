// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BS in CS: Interdisciplinary Program Node should have expected snapshot 1`] = `
{
  "internalId": "RG-11227",
  "key": "bs_in_cs___interdisciplinary_2019",
  "name": "BS In CS - Interdisciplinary 2019",
  "satisfied": "INCOMPLETE",
  "subNodes": [
    {
      "description": "Racial and Ethnic StudiesStudents must take 6 credits, a minimum of 3 credits must be from RES-A",
      "internalId": "RQ-10360",
      "isGenEd": true,
      "key": "res_requirement",
      "name": "RES Requirement",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "Global Perspectives",
      "internalId": "RQ-025",
      "isGenEd": true,
      "key": "glp_requirement",
      "name": "GLP Requirement",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education: Communication Skills",
      "internalId": "RQ-020",
      "isGenEd": true,
      "key": "gened__communication_skills",
      "name": "GenEd: Communication Skills",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education-Analytical Reasoning & Natural ScienceComputer Science Requirements",
      "internalId": "RQ-10588",
      "isGenEd": true,
      "key": "gened__arns",
      "name": "GenEd: ARNS",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education: Humanities and the Arts- Select from two or more areas Including: Art History/Music Appreciation, Creative/Performing Arts, History, Literature, Philosophy, & Foreign Language & Culture  *Effective 01/01/2021, PHOTO 130 NO LONGER classified as GE HUM/ART & does not satisfy this area.   **Effective 08/15/2021, ART 100 & ART 101 NO LONGER classified as GE HUM/ART & do not satisfy this area.",
      "internalId": "RQ-019",
      "isGenEd": true,
      "key": "gened__humart",
      "name": "GenEd: HumArt",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education: Social and Behavioral Sciences - Courses MUST be from two or more Areas Including: Anthropology, Economics, Geography, Political Science, Psychology and Sociology",
      "internalId": "RQ-0138",
      "isGenEd": true,
      "key": "gened__sbsci",
      "name": "GenEd: SBSci",
      "satisfied": "INCOMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education: Cross-Disciplinary Issues",
      "internalId": "RQ-10361",
      "isGenEd": true,
      "key": "gened__ci",
      "name": "GenEd: CI",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "General Education- Social Responsibility and Ethical ReasoningTake ICT 215",
      "internalId": "RQ-10432",
      "isGenEd": true,
      "key": "gened__srer",
      "name": "GenEd: SRER",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
    {
      "description": "COMPUTER SCIENCE MAJOR STUDIES A grade of C or better is required for all MATH, CS, and GDD courses. ",
      "internalId": "RQ-0451",
      "isGenEd": false,
      "key": "maj__cs_major_studies",
      "name": "Maj: CS Major Studies",
      "satisfied": "INCOMPLETE",
      "subNodes": [],
    },
    {
      "description": "Student must complete a minimum of 6 Natural Science Credits",
      "internalId": "RQ-010760",
      "isGenEd": false,
      "key": "verify_completion_of_natural_science_credit_requirement",
      "name": "Verify Completion Of Natural Science Credit Requirement",
      "satisfied": "INCOMPLETE",
      "subNodes": [],
    },
    {
      "description": "Interdisciplinary Concentration",
      "internalId": "RQ-10608",
      "isGenEd": false,
      "key": "cs___interdisciplinary_conc",
      "name": "CS - Interdisciplinary CONC",
      "satisfied": "INCOMPLETE",
      "subNodes": [],
    },
    {
      "description": "OTHER COURSES",
      "internalId": "RQ-0331",
      "isGenEd": false,
      "key": "other_courses",
      "name": "Other Courses",
      "satisfied": "COMPLETE",
      "subNodes": [],
    },
  ],
}
`;
